pkg/utils/utils.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"regexp"
+
+	"github.com/containernetworking/cni/pkg/types"
+)
+
+// ValidInputString is the regexp used to validate valid characters in
+// containerID and networkName
+const cniValidNameChars = `[a-zA-Z0-9][a-zA-Z0-9_.-]`
+
+var cniReg = regexp.MustCompile(`^` + cniValidNameChars + `+$`)
```

Comment:
This and above lines are introducing expectation/limitation that name should be constructed from two chars at least. Shouldn't you use there `*` in place of `+`? If `+` is correct - you should update also the spec with note about this limitation.

Reply:
Yes, "a" seems a good name for cni network name, but it can't pass regexp validation.

Reply:
Good catch @jellonek.